pkg/vault/test/foundry/fork/YieldBearingPoolSwapBase.t.sol

```diff
@@ -0,0 +1,1097 @@
+        // PoolMock is linear, so wrappedAmountInScaled18 = wrappedAmountOutScaled18
+        uint256 wrappedBufferDeltaTokenInRaw = wrappedBufferDeltaTokenOutScaled18.mulDown(_ybToken1Factor);
+        int256 expectedBufferDeltaTokenIn = int256(ybToken1.convertToAssets(wrappedBufferDeltaTokenInRaw));
+
+        int256 expectedBufferDeltaTokenOut = -int256(amountOut);
+
+        _testExactOut(paths, true, expectedBufferDeltaTokenIn, expectedBufferDeltaTokenOut);
+    }
+
+    function testToken1InToken2OutOutOfBufferExactIn__Fork__Fuzz(uint256 amountIn) public {
+        // Test from 2x to 4x of tokenIn's buffer to make sure it's out of both buffer ranges but yield-bearing pool
+        // has enough tokens to trade (5x buffer init amount).
+        amountIn = bound(amountIn, 2 * _token1BufferInitAmount, 4 * _token1BufferInitAmount);
+        IBatchRouter.SwapPathExactAmountIn[] memory paths = _buildExactInPaths(_token1Fork, amountIn, 0);
+
+        // Since operation is out of buffer range, buffer balances should not change (wrap/unwrap tokens directly).
+        int256 expectedBufferDeltaTokenIn = 0;
+        int256 expectedBufferDeltaTokenOut = 0;
+
+        _testExactIn(paths, false, expectedBufferDeltaTokenIn, expectedBufferDeltaTokenOut);
+    }
+
+    function testToken1InToken2OutOutOfBufferExactOut__Fork__Fuzz(uint256 amountOut) public {
+        // Test from 2x to 4x of tokenOut's buffer to make sure it's out of both buffer ranges but yield-bearing pool
+        // has enough tokens to trade (5x buffer init amount).
+        amountOut = bound(amountOut, 2 * _token2BufferInitAmount, 4 * _token2BufferInitAmount);
+        IBatchRouter.SwapPathExactAmountOut[] memory paths = _buildExactOutPaths(
+            _token1Fork,
+            (2 * amountOut * _token1Factor) / _token2Factor,
+            amountOut
+        );
+
+        // Since operation is out of buffer range, buffer balances should not change (wrap/unwrap tokens directly).
+        int256 expectedBufferDeltaTokenIn = 0;
+        int256 expectedBufferDeltaTokenOut = 0;
+
+        _testExactOut(paths, false, expectedBufferDeltaTokenIn, expectedBufferDeltaTokenOut);
+    }
+
+    function testToken1InToken2OutBufferUnbalancedExactIn__Fork__Fuzz(
+        uint256 amountIn,
+        uint256 unbalancedToken1,
+        uint256 unbalancedToken2
+    ) public {
+        unbalancedToken1 = bound(unbalancedToken1, 0, _token1BufferInitAmount);
+        unbalancedToken2 = bound(unbalancedToken2, 0, _token2BufferInitAmount);
+        _unbalanceBuffers(unbalancedToken1, unbalancedToken2);
+
+        amountIn = bound(amountIn, 2 * _token1BufferInitAmount, 4 * _token1BufferInitAmount);
+        IBatchRouter.SwapPathExactAmountIn[] memory paths = _buildExactInPaths(_token1Fork, amountIn, 0);
+
+        int256 expectedBufferDeltaTokenIn;
+        int256 expectedBufferDeltaTokenOut;
+
+        // If the difference is smaller than MIN_TRADE_AMOUNT, buffer is not unbalanced.
```

Comment:
Can we rephrase this a bit? This describes the `else` clause right?

I'd describe the `if` here (if the buffer is unbalanced beyond `MIN_TRADE_AMOUNT`, we expect a delta in the buffer.

Reply:
Hmm, good point.. I tried to explain the MIN_TRADE_AMOUNT variable, but you're right. I'll change this comment.